admin/src/containers/MainLayout/MainLayout.js

```diff
@@ -38,42 +40,51 @@ class MainLayoutContainer extends Component {
 
   render() {
     return (
-      <UsersProvider>
-        <PanelProvider>
-          <PanelConsumer>
-            {({ component: Component, title: panelTitle, props, closePanel, isPanelOpen }) => (
-              <MainLayout
-                isSidenavOpen={this.state.isSidenavOpen}
-                isPanelOpen={isPanelOpen}
-                username={session.getUsername()}
-                onMenuClick={this.toggleSidenav.bind(this)}
-                onOverlayClick={closePanel}
-                onClosePanelClick={closePanel}
-                panelContent={Component ? <Component {...props} /> : null}
-                panelTitle={panelTitle}
-                content={
-                  <Switch>
-                    <Route
-                      exact
-                      path="/"
-                      render={() => <React.Fragment>Dashboard</React.Fragment>}
-                    />
-                    <Route exact path="/processors" render={() => <ProcessorsContainer />} />
-                    <Route exact path="/users" render={() => <UsersContainer />} />
-                    <Route
-                      exact
-                      path="/profile"
-                      render={() => <React.Fragment>Profile</React.Fragment>}
-                    />
-                  </Switch>
-                }
-              />
-            )}
-          </PanelConsumer>
-        </PanelProvider>
-      </UsersProvider>
+      <PanelConsumer>
+        {({ component: Component, title: panelTitle, props, closePanel, isPanelOpen }) => (
+          <MainLayout
+            isSidenavOpen={this.state.isSidenavOpen}
+            isPanelOpen={isPanelOpen}
+            username={session.getUsername()}
+            onMenuClick={this.toggleSidenav.bind(this)}
+            onOverlayClick={closePanel}
+            onClosePanelClick={closePanel}
+            panelContent={Component ? <Component {...props} /> : null}
+            panelTitle={panelTitle}
+            content={
+              <Switch>
+                <Route exact path="/" render={() => <React.Fragment>Dashboard</React.Fragment>} />
+                <Route exact path="/processors" render={() => <ProcessorsContainer />} />
+                <Route exact path="/users" render={() => <UsersContainer />} />
+                <Route
+                  exact
+                  path="/profile"
+                  render={() => <React.Fragment>Profile</React.Fragment>}
+                />
+              </Switch>
+            }
+          />
+        )}
+      </PanelConsumer>
     );
   }
 }
 
-export default MainLayoutContainer;
+const withProvider = ContextProvider => Component => {
+  let wrapper = props => (
+    <ContextProvider>
+      <Component {...props}>{props.children}</Component>
+    </ContextProvider>
+  );
+  wrapper.propTypes = {
+    children: PropTypes.node
+  };
+  return wrapper;
+};
+
+const withProviders = lodash.flow.apply(null, [
```

Comment:
using apply instead of just calling `lodash.flow()` is a workaround for some prettier / eslint issue that would trigger a pre-commit hook which would then fail to compile. Super awkward, maybe we can fix this together tomorrow.